src/vs/workbench/contrib/terminal/node/terminalProcess.ts

```diff
@@ -136,10 +136,10 @@ export class TerminalProcess extends Disposable implements ITerminalChildProcess
 		return undefined;
 	}
 
-	private setupPtyProcess(shellLaunchConfig: IShellLaunchConfig, options: pty.IPtyForkOptions): void {
+	private async setupPtyProcess(shellLaunchConfig: IShellLaunchConfig, options: pty.IPtyForkOptions): Promise<void> {
 		const args = shellLaunchConfig.args || [];
 		this._logService.trace('IPty#spawn', shellLaunchConfig.executable, args, options);
-		const ptyProcess = pty.spawn(shellLaunchConfig.executable!, args, options);
+		const ptyProcess = (await import('node-pty')).spawn(shellLaunchConfig.executable!, args, options);
```

Comment:
Looks good, you can also change the import to `import type` to make sure we don't accidentally import the code again?

Reply:
nvm, I did it 🙂.